src/autoupdater.ts

```diff
@@ -331,6 +331,21 @@ export class AutoUpdater {
       return false;
     }
 
+    const readyStateFilter = this.config.pullRequestReadyState();
+    if (readyStateFilter !== 'all') {
+      ghCore.info('Checking PR ready state');
+
+      if (readyStateFilter === 'draft' && !pull.draft) {
+        ghCore.info('Pull request is not draft, skipping update.');
+        return false;
+      }
+
+      if (readyStateFilter === 'ready_for_review' && pull.draft) {
+        ghCore.info('Pull request is draft, skipping update.');
+        return false;
+      }
+    }
+
```

Comment:
It occurs to me that this check needs to move up before any `return true` code paths to work correctly.  I’ll make that change in the coming days (or you can @chinthakagodawita if you’re so inclined).

Reply:
Done.